feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt

```diff
@@ -67,27 +68,25 @@ internal data class Price(
 // The lasts two are englobed in []*, as they can be repeated 0 or n times.
 private val pattern: Pattern = Pattern.compile("(\\d+[[\\.,\\s]\\d+]*)")
 
-internal fun String.parsePriceUsingRegex(): Float? {
+internal fun String.parsePriceUsingRegex(): BigDecimal? {
     val matcher = pattern.matcher(this)
-    return if (matcher.find()) {
+    return matcher.takeIf { it.find() }?.let {
         val dirtyPrice = matcher.group()
-        var price: String
-        // 2 355 825.837
-        price = dirtyPrice.replace(" ", "")
-        val hasCommas = price.contains(",")
-        if (hasCommas) {
-            val numberOfCommas = price.length - price.replace(",", "").length
-            price = if (price.contains(".") || numberOfCommas > 1) {
-                // 1,000,000.00
-                price.replace(",", "")
+        var price = dirtyPrice.replace(" ", "")
+        val split = price.split(".", ",")
+        if (split.size != 1) {
+            // Assuming all prices we get have 2 decimal points
+            // Most currencies but Dirhan use 2 decimals. Amazon doesn't support Dirhan at the moment
+            price = if (split.last().length == 3) {
```

Comment:
do we have any way to identify right-to-left orientation?

Reply:
We can check the locale of the device, but prices are not formatted according to the locale of the device. For example, I tested a Live App Testing build with a user in Amazon ES (the device was in US English), and the price was coming as "1,10 €".

In any case, I don't think right-to-left matters for numbers right? I believe prices would still be represented the same. Currency might go on the right instead but the number itself is not right-to-left.

Reply:
You're right! TIL: they're [represented the same](https://qr.ae/pGLRjH). That's good to know! 


> In Arabic, it's the natural ordering from right to left: least significant digit first, most significant last.
> 
> Historically, the Western numbering system is derived from the Hindu-Arabic numbering system; popularized by Muslim mathematicians like [Muḥammad ibn Mūsā al-Khwārizmī](https://en.wikipedia.org/wiki/Mu%E1%B8%A5ammad_ibn_M%C5%ABs%C4%81_al-Khw%C4%81rizm%C4%AB). When these books were translated from Arabic, however, Europeans who had a left-to-right language, still maintained the Arabic numbering order. Why Europeans did this nobody knowes.